sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -613,6 +614,106 @@ function createDefaultRequestPolicyFactories(
   return factories;
 }
 
+export function createPipelineFromOptions(
+  pipelineOptions: InternalPipelineOptions,
+  authPolicyFactory?: RequestPolicyFactory
+) : ServiceClientOptions {
+  let requestPolicyFactories: RequestPolicyFactory[] = [];
+
+  const userAgentInfo: string[] = [];
+  if (pipelineOptions.userAgentOptions && pipelineOptions.userAgentOptions.userAgentPrefix) {
+    userAgentInfo.push(pipelineOptions.userAgentOptions.userAgentPrefix);
+  }
+
+  const defaultUserAgentInfo = getDefaultUserAgentValue();
+  if (userAgentInfo.indexOf(defaultUserAgentInfo) === -1) {
+    userAgentInfo.push(defaultUserAgentInfo);
+  }
```

Comment:
I would do this default user agent calculation only if a prefix was provided. Else, this entire logic is already present in the user agent policy.

```javascript
const userAgentInfo: string | undefined = undefined;
  if (pipelineOptions.userAgentOptions && pipelineOptions.userAgentOptions.userAgentPrefix) {
    userAgentInfo = pipelineOptions.userAgentOptions.userAgentPrefix;
    const defaultUserAgentInfo = getDefaultUserAgentValue();
    if (userAgentInfo.indexOf(defaultUserAgentInfo) === -1) {
         userAgentInfo += " " + defaultUserAgentInfo;
     }
  }
```

Reply:
Yep, you're right, fixed this in the latest commit